### PR TITLE
Generate HTML for section headers and synopsis

### PR DESCRIPTION
**Pull request by vitvakatu**
### Pull Request Description

Two tasks:
 - [Task link](https://www.pivotaltracker.com/story/show/184024127)
 - [Task link](https://www.pivotaltracker.com/story/show/184024148)

This PR implements the generation of HTML from Documentation IR for section headers and the Synopsis section.
The synopsis contains documentation of the type/module + a list of the type's constructors.

https://user-images.githubusercontent.com/6566674/212684680-d999b525-56c7-4952-8ccc-192989acdf33.mp4

### Important Notes

- Paddings are removed from the documentation panel because they are now implemented in the HTML generator, but it doesn't affect the looks much (documentation still looks awful). All other changes do not affect the current look of the component browser and are only shown in the demo scene.

https://user-images.githubusercontent.com/6566674/212685347-addb9204-8441-44be-8d8e-3c2626d77f77.mp4

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

**Review by wdanilo — APPROVED** [review]
Looks good! The only thing that is different than in design is that the icons are not the size of letters. If you look at the design and you zoom in, you'll see that the top and bottom edge of the icon is aligned with the top and baseline of letter.

**Comment by vitvakatu** [comment]
The PR is updated, now it contains not only headers with icons but also a synopsis section. @wdanilo please take a look.

File: app/gui/view/documentation/src/html.rs [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+
+// ==============
+// === Render ===
+// ==============
+
+/// Render entry documentation to HTML code with Tailwind CSS styles.
+pub fn render(docs: EntryDocumentation) -> String {
+    match docs {
+        EntryDocumentation::Placeholder(_) => String::from("Temporary placeholder"),
```
Comment (wdanilo) [inline app/gui/view/documentation/src/html.rs]: is it possible that someone ever sees this text?

**Review by wdanilo — APPROVED** [review]

File: app/gui/view/documentation/src/html.rs (hunk 1 above)
Comment (vitvakatu) [reply app/gui/view/documentation/src/html.rs]: Shouldn't happen at the moment, but I'll improve it to avoid possible confusion.

**Pull request merged**